odoo/fields.py

```diff
@@ -1798,14 +1800,17 @@ def write(self, records, value):
                 continue
             from_lang_value = old_translations.get(lang, old_translations.get('en_US'))
             translation_dictionary = self.get_translation_dictionary(from_lang_value, old_translations)
-            text2term = {self.get_text_content(term): term for term in new_terms}
+            text2terms = dict()
+            for term in new_terms:
+                text2terms.setdefault(SPACES.sub(' ', self.get_text_content(term)).strip(), []).append(term)
```

Comment:
Maybe we can move the logic
`SPACES.sub()` and `strip()`
to `get_text_content` which is only used while writing model_terms field